flink-streaming-java/src/main/java/org/apache/flink/streaming/api/transformations/BroadcastStateTransformation.java

```diff
@@ -53,111 +50,91 @@
 		super(name, outTypeInfo, parallelism);
-		this.nonBroadcastStream = checkNotNull(inputStream);
+		this.inputStream = checkNotNull(inputStream);
 		this.broadcastStream = checkNotNull(broadcastStream);
-		this.operatorFactory = checkNotNull(operatorFactory);
+		this.userFunction = userFunction;
+		this.broadcastStateDescriptors = broadcastStateDescriptors;
 
-		this.stateKeyType = keyType;
-		this.keySelector = keySelector;
-		updateManagedMemoryStateBackendUseCase(keySelector != null);
+		updateManagedMemoryStateBackendUseCase(false /* not keyed */);
 	}
 
 	public Transformation<IN2> getBroadcastStream() {
 		return broadcastStream;
 	}
 
 	public Transformation<IN1> getNonBroadcastStream() {
-		return nonBroadcastStream;
+		return inputStream;
 	}
 
-	public StreamOperatorFactory<OUT> getOperatorFactory() {
-		return operatorFactory;
+	public BroadcastProcessFunction<IN1, IN2, OUT> getUserFunction() {
+		return userFunction;
 	}
 
-	public TypeInformation<?> getStateKeyType() {
-		return stateKeyType;
+	public List<MapStateDescriptor<?, ?>> getBroadcastStateDescriptors() {
+		return broadcastStateDescriptors;
 	}
 
-	public KeySelector<IN1, ?> getKeySelector() {
-		return keySelector;
+	public ChainingStrategy getChainingStrategy() {
+		return chainingStrategy;
 	}
 
 	@Override
-	public void setChainingStrategy(ChainingStrategy strategy) {
-		this.operatorFactory.getChainingStrategy();
+	public void setChainingStrategy(ChainingStrategy chainingStrategy) {
+		this.chainingStrategy = checkNotNull(chainingStrategy);
 	}
 
 	@Override
 	public List<Transformation<?>> getTransitivePredecessors() {
 		final List<Transformation<?>> predecessors = new ArrayList<>();
 		predecessors.add(this);
-		predecessors.add(nonBroadcastStream);
+		predecessors.add(inputStream);
 		predecessors.add(broadcastStream);
 		return predecessors;
 	}
 
 	@Override
 	public List<Transformation<?>> getInputs() {
 		final List<Transformation<?>> predecessors = new ArrayList<>();
-		predecessors.add(nonBroadcastStream);
+		predecessors.add(inputStream);
 		predecessors.add(broadcastStream);
 		return predecessors;
 	}
 
 	// ------------------------------- Static Constructors -------------------------------
 
-	public static <IN1, IN2, OUT> BroadcastStateTransformation<IN1, IN2, OUT> forNonKeyedStream(
-			final String name,
-			final DataStream<IN1> nonBroadcastStream,
-			final BroadcastStream<IN2> broadcastStream,
-			final StreamOperatorFactory<OUT> operatorFactory,
-			final TypeInformation<OUT> outTypeInfo,
-			final int parallelism) {
-		return new BroadcastStateTransformation<>(
-				name,
-				checkNotNull(nonBroadcastStream).getTransformation(),
-				checkNotNull(broadcastStream).getTransformation(),
-				operatorFactory,
-				null,
-				null,
-				outTypeInfo,
-				parallelism);
-	}
-
-	public static <IN1, IN2, OUT> BroadcastStateTransformation<IN1, IN2, OUT> forKeyedStream(
+	public static <IN1, IN2, OUT> BroadcastStateTransformation<IN1, IN2, OUT> forStream(
```

Comment:
We can make the constructor `public` now and remove the static constructors from here and from the `keyed` alternative as now we have two separate classes.